superscore/client.py

```
@@ -319,17 +318,23 @@ def _build_snapshot(
                 child = self.backend.get(child)
             if isinstance(child, Parameter):
                 if child.readback is not None:
+                    edata = values[child.readback.pv_name] or EpicsData(data=None)
```

If the PV name isn't in `values`, won't this will raise an error rather than returning `EpicsData(data=None)`?  Changing to `values.get(child.readback.pv_name, EpicsData(data=None))` should do what we want in one line (but the backup `EpicsData` instance will be created every time).

---

Good catch.  To add onto this, our data gathering process inserts None for any values that cannot be found by EPICS.  This means we still need to check that the value is not None.  (None specifically, since empty strings or 0's are false-y in python_